src/main/java/com/labsynch/labseer/service/LotServiceImpl.java

```diff
@@ -80,52 +87,107 @@ public Lot updateLotWeight(Lot lot) {
 		Parent adoptiveParent = Parent.findParentsByCorpNameEquals(parentCorpName).getSingleResult();
+		Set isoSalts = saltForm.getIsoSalts();
+		isoSalts.size();
+		if(dryRun) {
+			Lot.entityManager().detach(queryLot);
+			SaltForm.entityManager().detach(saltForm);
+			Parent.entityManager().detach(adoptiveParent);
+		}
+
+		LotOrphanLevel lotOrphanLevel = lotService.checkLotOrphanLevel(queryLot);
+
+		Boolean deleteOriginalParentAfterReparent = false;
+		Long originalParentId = queryLot.getParent().getId();
+		if (lotOrphanLevel.equals(LotOrphanLevel.Parent)) {
+			deleteOriginalParentAfterReparent = true;
+		}
+		
 		// renumber lot -- just increment to next number
-		Integer newLotNumber = Lot.getMaxParentLotNumber(adoptiveParent) + 1;
+		Integer newLotNumber = predictReparentLotNumber(adoptiveParent, queryLot);
 		logger.debug("next lot number for adoptive parent: " + newLotNumber);
 
-		Lot queryLot = Lot.findLotsByCorpNameEquals(lotCorpName).getSingleResult();
+		reparentLotResponseDTO.setOriginalParentCorpName(queryLot.getParent().getCorpName());
 
 		// associate saltForm to new adoptive parent
-		SaltForm saltForm = queryLot.getSaltForm();
 		saltForm.setParent(adoptiveParent);
 		saltForm.setCorpName(
-				corpNameService.generateSaltFormCorpName(adoptiveParent.getCorpName(), saltForm.getIsoSalts()));
-		saltForm.merge();
+				corpNameService.generateSaltFormCorpName(adoptiveParent.getCorpName(), isoSalts));
 
-		// recalculate salt form weight
-		saltFormService.updateSaltWeight(saltForm);
-
-		logger.debug("new lot number before merge: " + newLotNumber);
 		queryLot.setLotNumber(newLotNumber);
 		logger.debug("new lot number: " + queryLot.getLotNumber());
 
 		// rename lot corp name if not cas style
 		if (!propertiesUtilService.getCorpBatchFormat().equalsIgnoreCase("cas_style_format")) {
-			queryLot.setCorpName(this.generateCorpName(queryLot));
+			String newCorpName = this.generateCorpName(queryLot);
+			if(Lot.findLotsByCorpNameEquals(newCorpName).getResultList().size() > 0){
+				logger.error("Lot corp name already exists: " + newCorpName);
+				throw new DupeLotException("Lot corp name already exists: " + newCorpName);
+			}
+			queryLot.setCorpName(newCorpName);
 			// TODO: may want to save the old name as an alias
-			// also need to deal with data that may be registered to the old corp name
 		}
 		logger.info("new lot corp name: " + queryLot.getCorpName());
+
 		// recalculate lot weight
 		queryLot.setLotMolWeight(Lot.calculateLotMolWeight(queryLot));
 		queryLot.setModifiedBy(modifiedByUser);
 		queryLot.setModifiedDate(new Date());
-		queryLot.merge();
 
-		return queryLot;
+		reparentLotResponseDTO.setNewLot(queryLot);
+
+		if(deleteOriginalParentAfterReparent) {
+			reparentLotResponseDTO.setOriginalParentDeleted(true);
+		} else {
+			reparentLotResponseDTO.setOriginalParentDeleted(false);
+		}
+
+		if(!dryRun) {
+			saltForm.merge();
+			saltFormService.updateSaltWeight(saltForm);
```

Comment:
It's strange to me that this is called _after_ you `merge` the saltForm. Have you double checked that the updated Salt Weight actually goes through properly?

Reply:
Thanks, fixed this. Would have been a bug too if we used iso salts.

Reply:
I didn't rerun the test after my update to fix this and found that tests were failing. Looking into it.

Reply:
I found that `saltFormService.updateSaltWeight` was trying to call `saltForm.merge()`. I removed that call in ` `saltFormService.updateSaltWeight` and added it to the only time it was being used in LotServiceImpl.java on the `updateLotWeight` method.

Reply:
Tests now pass again